### PR TITLE
Use a backend create function to initialize the task module

### PR DESCRIPTION
**Pull request by madhuravi**

### CODE REVIEW

**Comment by madhuravi** [comment]
@uber-go/service-framework

File: modules/task/README.md [hunk 1]
```diff
@@ -23,9 +23,17 @@ import (
...
+    task.NewModule(newBackend),
   ).Build()
-  task.Register(updateCache)
+  svc.Start()
+  if err := task.Register(updateCache); err != nil {
+    // handle error
+  }
+}
+
+func newBackend(host service.Host) (task.Backend, error) {
```
Comment (anuptalwalkar) [inline modules/task/README.md]: Not sure why this needs to be in README.

File: modules/task/task.go [hunk 2]
```diff
@@ -36,6 +36,7 @@ var (
 	_globalBackendMu sync.RWMutex
 	_globalBackend   Backend = &NopBackend{}
 	_asyncMod        service.Module
+	_asyncModErr     error
```
Comment (anuptalwalkar) [inline modules/task/task.go]: Why is this global? You can just have it locally on line 53.

Comment (madhuravi) [inline modules/task/task.go]: Need it to be global so every time the function is called, it returns the module and error from the very first call.

File: modules/task/README.md (hunk 1 above)
Comment (madhuravi) [inline modules/task/README.md]: I was torn about this too. What do you suggest instead?

**Pull request merged**